src/transformState.js

```diff
@@ -5,7 +5,34 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'clear':
+        clearProperties(state);
+        break;
+      case 'addProperties':
+        Object.assign(state, extraData);
+        break;
+      case 'removeProperties':
+        removeKey(keysToRemove, state);
+        break;
+      default:
```

Comment:
would be nice to add some empty lines between different cases for improving readability
```suggestion
    switch (type) {
      case 'clear':
        clearProperties(state);
        break;
        
      case 'addProperties':
        Object.assign(state, extraData);
        break;
        
      case 'removeProperties':
        removeKey(keysToRemove, state);
        break;
        
      default:
```